packages/flutter/lib/src/painting/text_style.dart

```diff
@@ -429,112 +429,159 @@ class TextStyle {
+      '${prefix}weight',
+      fontWeight,
+      description: weightDescription,
+      defaultValue: null,
+    ));
+    styles.add(new EnumProperty<FontStyle>(
+      '${prefix}style',
+      fontStyle,
+      defaultValue: null,
+    ));
+    styles.add(new DoubleProperty(
+      '${prefix}letterSpacing',
+      letterSpacing,
+      unit: 'x',
+      defaultValue: null,
+    ));
+    styles.add(new DoubleProperty(
+      '${prefix}wordSpacing',
+      wordSpacing,
+      unit: 'x',
+      defaultValue: null,
+    ));
+    styles.add(new EnumProperty<TextBaseline>(
+      '${prefix}baseline',
+      textBaseline,
+      defaultValue: null,
+    ));
+    styles.add(new DoubleProperty(
+      '${prefix}height',
+      height, unit: 'x',
+      defaultValue: null,
+    ));
     if (decoration != null || decorationColor != null || decorationStyle != null) {
-      String decorationDescription = '${prefix}decoration: ';
-      bool haveDecorationDescription = false;
-      if (decorationStyle != null) {
-        switch (decorationStyle) {
-          case TextDecorationStyle.solid:
-            decorationDescription += 'solid';
-            break;
-          case TextDecorationStyle.double:
-            decorationDescription += 'double';
-            break;
-          case TextDecorationStyle.dotted:
-            decorationDescription += 'dotted';
-            break;
-          case TextDecorationStyle.dashed:
-            decorationDescription += 'dashed';
-            break;
-          case TextDecorationStyle.wavy:
-            decorationDescription += 'wavy';
-            break;
-        }
-        haveDecorationDescription = true;
-      }
-      if (decorationColor != null) {
-        if (haveDecorationDescription)
-          decorationDescription += ' ';
-        decorationDescription += '$decorationColor';
-        haveDecorationDescription = true;
-      }
-      if (decoration != null) {
-        if (haveDecorationDescription)
-          decorationDescription += ' ';
-        decorationDescription += '$decoration';
-        haveDecorationDescription = true;
-      }
-      assert(haveDecorationDescription);
-      result.add(decorationDescription);
+      final List<String> decorationDescription = <String>[];
+      if (decorationStyle != null)
+        decorationDescription.add(describeEnum((decorationStyle)));
+
+      // Hide decorationColor from the default text view as we show it in the
+      // terse decoration summary as well.
+      styles.add(new DiagnosticsProperty<Color>(
+        '${prefix}decorationColor',
+        decorationColor,
+        defaultValue: null,
+        hidden: true,
+      ));
+
+      if (decorationColor != null)
+        decorationDescription.add('$decorationColor');
+
+      // Intentionally collide with the property 'decoration' added below.
+      // Tools that show hidden properties could choose the first property
+      // matching the name to disambiguate.
```

Comment:
this case is ugly. the property called 'decoration' in the output depends on more than just the actual decoration property. is there another name for that union of 'decoration' and other stuff that would be better?